backend/src/db/entity-migrations/v3.1.1-add-certificates-table.ts

```diff
@@ -0,0 +1,24 @@
+import { Knex } from 'knex';
+import { CERTIFICATES_TABLE } from '../../repository/certificate.repository';
+
+export async function up(knex: Knex): Promise<void> {
+  await createCertificatesTable(knex);
+}
+
+async function createCertificatesTable(knex: Knex): Promise<void> {
+  await knex.schema.raw(`create table ${CERTIFICATES_TABLE}
+                                  (id          char(36) primary key,
+                                   created_at  datetime default CURRENT_TIMESTAMP not null,
+                                   updated_at  datetime default CURRENT_TIMESTAMP not null,
+                                   name        varchar not null,
+                                   description varchar,
+                                   public_key varchar not null,
+                                   private_key varchar not null,
+                                   private_key_passphrase varchar,
+                                   expiry varchar,
```

Comment:
Sometimes, it is useful to add the CA associated with the certificate, (see south MQTT for example)
Should we add an optional field for that ?

Reply:
how is a CA expressed in the form of another certificate stored in pem format?
do we need a root CA as well or a concatenation of both?

Reply:
When authenticating for MQTT with cert, the root CA is needed alongside the auth certificate. 
Other use cases (such as MongoDB I think) use a different approach where we need to concatenate both in one single cert.

That's why I suggest an optional field with root CA for these cases. That being said, it can be done later as an enhancement where the certificate service will be used for other protocols (such as MQTT, OPCUA...)

Reply:
but then that would not be for self signed certificate would it. It would be for manually imported certificate where we don't have a service and ui yet.